File: examples/go-vuln-demo/src/tp_true_positives.go
package main

import (
	"database/sql"
	"fmt"
	"os/exec"
)


func vulnerableQuery(db *sql.DB, userId string) (*sql.Row, error) {
	query := fmt.Sprintf("SELECT * FROM users WHERE id = %s", userId)
	return db.QueryRow(query), nil
}


func vulnerableExec(input string) (string, error) {
	out, err := exec.Command("sh", "-c", "echo "+input).Output()
	return string(out), err
}

File: examples/java-vuln-demo/src/com/secguardian/demo/tp_true_positives.java
package com.secguardian.demo;

// TP — True Positives: 真实漏洞，必须检出
import java.sql.*;

public class tp_true_positives {


    public void vulnerableQuery(Connection conn, String userId) throws SQLException {
        String sql = "SELECT * FROM users WHERE id = " + userId;  // 拼接！
        try (Statement stmt = conn.createStatement()) {
            ResultSet rs = stmt.executeQuery(sql);
        }
    }


    public void vulnerableExec(String input) throws Exception {
        String cmd = "ping " + input;  // 拼接！
        Runtime.getRuntime().exec(cmd);
    }
}
